src/pt/state.jl

```diff
@@ -40,9 +40,14 @@ discrete_variables(state::Union{Nothing, Pigeons.StreamState}) = []
 
 continuous_variables(state::Array) = SINGLETON_VAR
 discrete_variables(state::Array) = []
+
 function update_state!(state::Array, name::Symbol, index, value) 
-    state[name][index] = value
```

Comment:
Dangerous, could silently fail. Instead, @assert that `name === :singleton_variable`